src/site/filters/liquid.js

```diff
@@ -771,13 +771,15 @@ module.exports = function registerFilters() {
     });
   };
 
-  //* Sorts event dates (fieldDatetimeRangeTimezone) from newest to oldest.
-  liquid.filters.eventDateSorter = dates => {
+  //* Sorts event dates (fieldDatetimeRangeTimezone) starting with the most upcoming event.
+  //* Also sorts press releases (fieldReleaseDate) from newest to oldest.
+  liquid.filters.eventOrPressReleasesDateSorter = (dates, dateKey, reverse) => {
```

Comment:
This function could be used to sort anything with dates. How about calling it just `sortByDateKey`?

Also, `dateKey` is now a required field. You might want to use `fieldDatetimeRangeTimezone` as the default.

```suggestion
  liquid.filters.sortByDateKey = (dates, dateKey = 'fieldDatetimeRangeTimezone', reverse = false) => {
```

Reply:
We should probably make this function accept an object rather than a list of params, so you can pass any combination in any order.

This is generally a good idea once a function starts getting above 2 params, esp. if they aren't all required.

Reply:
This is true, I can rename.

Reply:
Hmm I'm not so sure about passing `fieldDatetimeRangeTimezone` as a default value. In my opinion, if we set a default value, it should generally always work. In this case, `fieldDatetimeRangeTimezone` will work for events but will not work for sorting press releases. What are your thoughts about it?

Reply:
Just because it doesn't always work, doesn't mean we shouldn't have a default. `fieldDatetimeRangeTimezone` appears in many places, including `src/site/includes/date.drupal.liquid`, so it seems like a good default value to me.

Reply:
Having a default value implies that that argument is optional to pass in - meaning that if I don't pass anything in, it will run as intended every time. I guess the way I am seeing it is, we are only using `sortByDateKey` in 2 places where the key passed in can be either `fieldDatetimeRangeTimezone ` for events or `fieldReleaseDate` for press releases.

I can add the default though 😁, maybe I'm just missing something

Reply:
> Having a default value implies that that argument is optional to pass in - meaning that if I don't pass anything in, it will run as intended every time.

Not necessarily. Setting a parameter default just gives the parameter a default value, which I think will be the same most of the time. It doesn't guarantee that the function will "run as intended every time".

I anticipate we might use this function in more places over time as we continue to migrate code out of the build.

But this is a minor point. We can always add the default later.

Reply:
thank you for your explanation and thoughts on it, I have added the default value.